src/main/java/io/cdap/e2e/pages/locators/CdfConnectionLocators.java

```diff
@@ -38,6 +38,11 @@ public class CdfConnectionLocators {
   @FindBy(how = How.XPATH, using = "//*[@data-cy='connection-browser-search']//input")
   public static WebElement searchDirectoryInput;
 
+  public static WebElement locateSelectButtonOfConnectionDataRow(String dataName) {
+    return SeleniumDriver.getDriver().findElement(
+      By.xpath("//*[text()='" + dataName + "']/ancestor::a//button"));
```

Comment:
Can the locator be more specific to only locate `Select` button. This might be problematic if more buttons are added in future in the same row in UI?

Reply:
Updated locator